api/server/runner_httptrigger_test.go

```diff
@@ -0,0 +1,622 @@
+		t.Errorf("Test %d: GetLog for call_id:%s returned err %s",
+			tnum, callID, err.Error())
+		return false
+	}
+
+	logBytes, err := ioutil.ReadAll(logReader)
+	if err != nil {
+		t.Errorf("Test %d: GetLog read IO call_id:%s returned err %s",
+			tnum, callID, err.Error())
+		return false
+	}
+
+	logBody := string(logBytes)
+	maxLog := len(logBody)
+	if maxLog > 1024 {
+		maxLog = 1024
+	}
+
+	for _, match := range expected {
+		if !strings.Contains(logBody, match) {
+			t.Errorf("Test %d: GetLog read IO call_id:%s cannot find: %s in logs: %s",
+				tnum, callID, match, logBody[:maxLog])
+			return false
+		}
+	}
+
+	return true
+}
+
+// implement models.MQ and models.APIError
+type errorMQ struct {
+	error
+	code int
+}
+
+func (mock *errorMQ) Push(context.Context, *models.Call) (*models.Call, error) { return nil, mock }
+func (mock *errorMQ) Reserve(context.Context) (*models.Call, error)            { return nil, mock }
+func (mock *errorMQ) Delete(context.Context, *models.Call) error               { return mock }
+func (mock *errorMQ) Code() int                                                { return mock.code }
+func (mock *errorMQ) Close() error                                             { return nil }
+func TestFailedEnqueue(t *testing.T) {
+	buf := setLogBuffer()
+	app := &models.App{ID: "app_id", Name: "myapp", Config: models.Config{}}
+	ds := datastore.NewMockInit(
+		[]*models.App{app},
+		[]*models.Route{
+			{Path: "/dummy", Image: "dummy/dummy", Type: "async", Memory: 128, Timeout: 30, IdleTimeout: 30, AppID: app.ID},
+		},
+	)
+	err := errors.New("Unable to push task to queue")
+	mq := &errorMQ{err, http.StatusInternalServerError}
+	fnl := logs.NewMock()
+	rnr, cancelrnr := testRunner(t, ds, mq, fnl)
+	defer cancelrnr()
+
+	srv := testServer(ds, mq, fnl, rnr, ServerTypeFull)
+	for i, test := range []struct {
+		path            string
+		body            string
+		method          string
+		expectedCode    int
+		expectedHeaders map[string][]string
+	}{
+		{"/r/myapp/dummy", ``, "POST", http.StatusInternalServerError, nil},
+	} {
+		body := strings.NewReader(test.body)
+		_, rec := routerRequest(t, srv.Router, test.method, test.path, body)
+		if rec.Code != test.expectedCode {
+			t.Log(buf.String())
+			t.Errorf("Test %d: Expected status code to be %d but was %d",
+				i, test.expectedCode, rec.Code)
+		}
+	}
+}
+
+func TestTriggerRunnerGet(t *testing.T) {
+	buf := setLogBuffer()
+	app := &models.App{ID: "app_id", Name: "myapp", Config: models.Config{}}
+	ds := datastore.NewMockInit(
+		[]*models.App{app},
+	)
+
+	rnr, cancel := testRunner(t, ds)
+	defer cancel()
+	logDB := logs.NewMock()
+	srv := testServer(ds, &mqs.Mock{}, logDB, rnr, ServerTypeFull)
+
+	for i, test := range []struct {
+		path          string
+		body          string
+		expectedCode  int
+		expectedError error
+	}{
+		{"/t/app/route", "", http.StatusNotFound, models.ErrAppsNotFound},
+		{"/t/myapp/route", "", http.StatusNotFound, models.ErrTriggerNotFound},
```

Comment:
No success test?

Reply:
Yep -about 30 of them further down the file